src/themes/mixins.js

```diff
@@ -0,0 +1,174 @@
+// Let's start by generating a type scale, along with line-heights and small-caps modifiers.
+// These functions will be used by our type styles below.
+
+// In order to get a whole-integer pixel number, we multiply by ten and round,
+// then divide by ten again to get a rem value that works with our base size setting.
+const calculateFontSize = ({ starterSize, scaleModifier, sizeNumber }) => {
+  const newSize =
+    Math.round(starterSize * 10 * scaleModifier ** sizeNumber) / 10;
+  return newSize;
+};
+
+// Calculate line height for either single-line (heading) or multi-line (body) usage.
+const calculateLineHeight = ({ fontSize, targetLineHeight }) => {
+  // Now we check to see if the font size * line-height is divisible by 4
+  // If not, increment until it is divisible by four.
+  // This ensures that line-height respects our vertical rhythm.
+  let targetLineHeightInPixels = Math.ceil(targetLineHeight * fontSize);
+
+  // Increment the target line height until it's a multiple of 4
+  while (targetLineHeightInPixels % 4 !== 0) {
+    targetLineHeightInPixels += 1;
+  }
+
+  // Generate a unitless line height relative to the font size
+  // that corresponds with our target pixel height.
+  const lineHeight = targetLineHeightInPixels / fontSize;
+  return lineHeight;
+};
+
+// Output typographic styling to apply to elements.
+const outputTypeDetails = ({ theme, sizeNumber, font }) => {
+  const fontSize = calculateFontSize({
+    starterSize: theme.typography.starterSizes.desktop,
+    scaleModifier: theme.typography.scaleModifiers.desktop,
+    sizeNumber,
+  });
+
+  // Get the target line height from our theme for the font used.
+  let targetLineHeight;
+  if (font === 'headings') {
+    targetLineHeight = theme.typography.lineHeights.headings;
+  } else if (font === 'interface') {
+    targetLineHeight = theme.typography.lineHeights.interface;
+  } else {
+    targetLineHeight = theme.typography.lineHeights.body;
+  }
+
+  return {
+    fontSize: `${fontSize}rem`,
+    lineHeight: calculateLineHeight({
+      fontSize: fontSize * 10,
+      targetLineHeight,
+    }),
+  };
+};
+
+// Convert regular case to small-caps.
+const smallCaps = () => {
+  return {
+    letterSpacing: '0.05em',
+    textTransform: 'uppercase',
+  };
+};
+
+// Interface
+export const interfaceSmall = (theme) => {
+  return {
+    fontFamily: theme.typography.fonts.interface,
+    fontWeight: theme.typography.fontWeights.interface,
+    ...outputTypeDetails({ theme, sizeNumber: -0.5, font: 'interface' }),
+  };
+};
+
+export const interfaceMedium = (theme) => {
+  return {
+    fontFamily: theme.typography.fonts.interface,
+    fontWeight: theme.typography.fontWeights.interface,
+    ...outputTypeDetails({ theme, sizeNumber: 0, font: 'interface' }),
```

Comment:
Should `'interface'` be a variable? Or is it a font we describe elsewhere and this just magically works?

Reply:
Oh, I see, they're keys for the `fonts` section in the theme. I'd prefer these used some kind of variable rather than these magic constants everywhere. I'll try to tidy them up.

Reply:
(Got it sorted 👍)

Reply:
❤️❤️❤️